Update ERC-7417: Move to Review (#658)
ethereum/ERCs: ERCS/erc-7417.md

As Ethereum already has an established ecosystem of tokens and [ERC-20](./eip-20.md) is the most adopted standard at the moment the lack of defined migration processes can be a bottleneck for newer standards adoption. This proposal addresses the problem of coordinating the upgrading process and addresses the backwards compatibility problems for [ERC-20](./eip-20.md) and [ERC-223](./eip-223.md) tokens.

The Token Converter is supposed to allow anyone to create an alternative version of an existing token implemented in a different standard. This proposal focuses on [ERC-20](./eip-20.md) and [ERC-223](./eip-223.md) standards and takes into account the specifics of this particular token standards. It is assumed that the most common case would be creation of [ERC-223](./eip-223.md) version for an existing [ERC-20](./eip-20.md) token.

The implementation of this service is an alternative to convincing each token developer to choose an alternative standard at the moment of the token deployment or during the development stage of their project. With this service there will be no need to choose one standard and stick with it as every token can be available in both concurrently.

> The implementation of this Token Converter service is supposed to be a contract deployed on Ethereum mainnet once and forever. It's address will be provided in the text of this proposal as to avoid any potential trust issues and assure the developers that the service they are interacting with is exactly the one which drives the conversion process of existing tokens.

Reviewer: > It's address will be provided in the text of this proposal [...]

Should be written in the present tense:

```suggestion
The implementation of this Token Converter service is supposed to be a contract deployed on Ethereum mainnet once and forever. Its address is provided in the text of this proposal as to avoid any potential trust issues and assure the developers that the service they are interacting with is exactly the one which drives the conversion process of existing tokens.
```

All the [ERC-223](./eip-223.md) tokens created by the Token Converter will be identical in a way that they all implement the same functions, which return the same values and there is no ambiguity there. This helps to avoid problems where a token deployed during the early stage of a token standard adoption may implement it improperly or there can be an ambiguity in the standard itself that would allow developers to implement tokens of one standard in different ways. 

For example it was a common case with [ERC-20](./eip-20.md) where developers could implement custom logic of the `transfer` function and mess the return values. The [ERC-20](./eip-20.md) specification declares that a `transfer` function MUST return a `bool` value, however in practice we have three different types of [ERC-20](./eip-20.md) tokens which are not compatible with each other:

1. [ERC-20](./eip-20.md) tokens that return `true` on success and revert on an error.
2. [ERC-20](./eip-20.md) tokens that return `true` on success and `false` on an error without reverting the transaction.
3. [ERC-20](./eip-20.md) tokens that don't have return values and revert on an error.

> Technically the third category of tokens is not compatible with [ERC-20](./eip-20.md) standard. However, USDT token deployed on Ethereum mainnet at `0xdac17f958d2ee523a2206206994597c13d831ec7` address does not implement return values and it is one of the most used tokens and it is not an option to deny supporting USDT due to it's improper implementation of the standard.

Reviewer: ```suggestion
Technically the third category of tokens is not compatible with [ERC-20](./eip-20.md) standard. However, USDT token deployed on Ethereum mainnet at `0xdac17f958d2ee523a2206206994597c13d831ec7` address does not implement return values and it is one of the most used tokens and it is not an option to deny supporting USDT due to its improper implementation of the standard.
```
